File: Searching/Linear Search/java/linearSearch.java
class LinearSearch 
{ 
    // This function returns index of element x in arr[] 
    static int search(int arr[], int n, int x) 
    { 
        for (int i = 0; i < n; i++) 
        { 
            // Return the index of the element if the element 
            // is found 
            if (arr[i] == x) 
                return i; 
        } 
   
        // return -1 if the element is not found 
        return -1; 
    } 
}  

File: Searching/Linear Search/LinearSearch.cs
using System;

public class Program
{
	public void Main()
	{
		var arrayToSearch = new int[]{1, 2, 3, 4, 5};
		// Should match at index 2
		Console.WriteLine(LinearSearch(arrayToSearch, 3));
		// 7 does not exist in the array to search, will return -1
		Console.WriteLine(LinearSearch(arrayToSearch, 7));
	}

	/// <summary>
	/// Searches through the arrayToSearch to find the target.
	///	Returns the index the target was found at, or -1 if no match was found
	/// </summary>
	public static int LinearSearch(int[] arrayToSearch, int target)
	{
		for (var i = 0; i < arrayToSearch.Length; i++)
		{
			if (arrayToSearch[i] == target)
				return i;
		}

		return -1;
	}
}